python/cuml/cuml/linear_model/elastic_net.py

```diff
@@ -209,156 +196,91 @@ def _attrs_to_cpu(self, model):
-
         self.alpha = alpha
         self.l1_ratio = l1_ratio
         self.fit_intercept = fit_intercept
-        self.solver = solver
-        self.normalize = normalize
         self.max_iter = max_iter
         self.tol = tol
-        self.solver_model = None
-        if selection in ["cyclic", "random"]:
-            self.selection = selection
-        else:
-            msg = "selection {!r} is not supported"
-            raise TypeError(msg.format(selection))
+        self.solver = solver
+        self.selection = selection
+        self.normalize = normalize
 
-        self.intercept_value = 0.0
+    @generate_docstring()
+    def fit(
+        self, X, y, sample_weight=None, *, convert_dtype=True
+    ) -> "ElasticNet":
+        """
+        Fit the model with X and y.
 
-        shuffle = False
-        if self.selection == "random":
-            shuffle = True
+        """
+        if self.alpha < 0.0:
+            raise ValueError(f"Expected alpha >= 0, got {self.alpha}")
+        if self.selection not in ["cyclic", "random"]:
+            raise ValueError(f"selection {self.selection!r} is not supported")
+        if self.l1_ratio < 0.0 or self.l1_ratio > 1.0:
+            raise ValueError(
+                f"Expected 0.0 <= l1_ratio <= 1.0, got {self.l1_ratio}"
+            )
 
-        if solver == "qn":
-            pams = signature(self.__init__).parameters
-            if pams["selection"].default != selection:
-                warn(
-                    "Parameter 'selection' has no effect "
-                    "when 'qn' solver is used."
-                )
-            if pams["normalize"].default != normalize:
-                warn(
-                    "Parameter 'normalize' has no effect "
-                    "when 'qn' solver is used."
+        if self.solver == "qn":
+            if self.normalize:
+                raise ValueError(
+                    "`normalize=True` is not supported with `solver='qn'"
                 )
 
-            self.solver_model = QN(
+            solver = QN(
+                handle=self.handle,
+                verbose=self.verbose,
+                output_type=self.output_type,
                 fit_intercept=self.fit_intercept,
                 l1_strength=self.alpha * self.l1_ratio,
                 l2_strength=self.alpha * (1.0 - self.l1_ratio),
-                max_iter=self.max_iter,
-                handle=self.handle,
                 loss="l2",
-                tol=self.tol,
                 penalty_normalized=False,
-                verbose=self.verbose,
+                max_iter=self.max_iter,
+                tol=self.tol,
+            ).fit(
+                X, y, sample_weight=sample_weight, convert_dtype=convert_dtype
             )
-        elif solver == "cd":
-            self.solver_model = CD(
-                fit_intercept=self.fit_intercept,
-                normalize=self.normalize,
+
+            coef = CumlArray(data=solver.coef_.to_output("cupy").flatten())
```

Comment:
Does the `CumlArray` saves a reference to the `coef_` attribute of the solver? I think it should, but not 100% certain.